overlays/nodejs/lambda/lambdaMixins.ts

```diff
@@ -0,0 +1,308 @@
+
+/**
+ * FunctionOptions provides configuration options for the serverless Function.  It is effectively
+ * equivalent to [aws.lambda.FunctionArgs] except with a few important differences documented at the
+ * property level.  For example, [role] is an actual iam.Role instance, and not an ARN. Properties
+ * like [runtime] are now optional.  And some properties (like [code]) are entirely disallowed.
+ */
+export type FunctionOptions<E, R> = utils.Overwrite<lambdaFunction.FunctionArgs, {
+    /**
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
+     */
+    code?: never;
+
+    /**
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
+     */
+    handler?: never;
+
+    /**
+     * The Javascript function instance to use as the entrypoint for the AWS Lambda out of.  Either
+     * [func] or [factoryFunc] must be provided.
+     */
+    func?: EntryPoint<E, R>;
+
+    /**
+     * The Javascript function instance that will be called to produce the function that is the
+     * entrypoint for the AWS Lambda. Either [func] or [factoryFunc] must be provided.
+     *
+     * This form is useful when there is expensive initialization work that should only be executed
+     * once.  The factory-function will be invoked once when the final AWS Lambda module is loaded.
+     * It can run whatever code it needs, and will end by returning the actual function that Lambda
+     * will call into each time the Lambda is invoked.
+     */
+    factoryFunc?: EntryPointFactory<E, R>;
+
+    /**
+     * A pre-created role to use for the Function. If not provided, [policies] will be used.
+     */
+    role?: iam.Role;
+
+    /**
+     * A list of IAM policy ARNs to attach to the Function.  Will be used if [role] is not provide.
+     * If neither [role] nor [policies] is provided, a default policy of [iam.AWSLambdaFullAccess]
+     * will be used instead.
+     */
+    policies?: arn.ARN[];
+
+    /**
+     * The Lambda runtime to use.  If not provided, will default to [NodeJS8d10Runtime]
+     */
+    runtime?: runtime.Runtime;
+
+    /**
+     * The paths relative to the program folder to include in the Lambda upload.  Default is `[]`.
+     */
+    includePaths?: string[];
+
+    /**
+     * The packages relative to the program folder to include in the Lambda upload.  The version of
+     * the package installed in the program folder and it's dependencies will all be included.
+     * Default is `[]`.
+     */
+    includePackages?: string[];
+
+    /**
+     * The packages relative to the program folder to not include the Lambda upload. This can be
+     * used to override the default serialization logic that includes all packages referenced by
+     * project.json (except @pulumi packages).  Default is `[]`.
+     */
+    excludePackages?: string[];
```

Comment:
Note: i would like to take includePaths/includePackage/excludePackages and make a new CodePathOptions type for them.  That way, if we need to ever add/change options to pulumi.computeCodePaths, we can just update that type (and reference it here), instead of needing to downstream flow all those changes.